server/service/src/stocktake/insert.rs

```diff
@@ -85,18 +85,23 @@ fn generate(
         description,
         stocktake_date,
         is_locked,
-        location_id: _,
+        location_id,
         master_list_id,
     }: InsertStocktake,
 ) -> Result<(StocktakeRow, Vec<StocktakeLineRow>), RepositoryError> {
     let stocktake_number = next_number(connection, &NumberRowType::Stocktake, store_id)?;
 
-    let lines = match master_list_id {
+    let master_list_lines = match master_list_id {
         Some(master_list_id) => {
             generate_lines_from_master_list(connection, store_id, &id, &master_list_id)?
         }
         None => Vec::new(),
     };
+    let location_lines = match location_id {
+        Some(location_id) => generate_lines_from_location(connection, store_id, &id, &location_id)?,
+        None => Vec::new(),
+    };
+    let lines = [master_list_lines, location_lines].concat();
```

Comment:
I am a little bit concerned that this could create stocktake with duplicate stocklines (not sure what the behaviour for this would be).   

I think this PR can be merged but with an issue to flag this

Reply:
true, I'll add an input check as well to prevent sending both location and list.